.github/workflows/dockerhub.yml

```diff
@@ -1,228 +1,311 @@
 name: Docker Hub build
+
 on:
   push:
     branches:
-      - "develop"
-      - "master"
+      - develop
+      - master
       - "release/*"
+    tags:
+      - "carma-system-*"
 
 jobs:
+  determine_docker_org_and_tag:
+    runs-on: ubuntu-latest
+    outputs:
+      docker_organization: ${{ steps.docker-org-and-tag.outputs.docker_organization }}
+      docker_image_tag: ${{ steps.docker-org-and-tag.outputs.docker_image_tag }}
+    steps:
+      - id: docker-org-and-tag
+        uses: usdot-fhwa-stol/actions/docker-org-and-tag@main
+
   dockerhub-telematic-cloud-messaging:
+    needs: determine_docker_org_and_tag
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+      - name: Get Docker Image Metadata
+        shell: bash
+        id: docker-image-metadata
+        run: |
+          echo build_date=$(date -u +”%Y-%m-%dT%H:%M:%SZ”) >> $GITHUB_OUTPUT
+          echo vcs_ref=$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+        uses: docker/setup-buildx-action@v3
       - name: Login to DockerHub
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build
-        uses: docker/build-push-action@v3
+      - name: Build and Push Docker Image
+        uses: docker/build-push-action@v5
         with:
           context: ./telematic_system/telematic_cloud_messaging/
-          file: ./telematic_system/telematic_cloud_messaging/Dockerfile
           push: true
-          tags: usdotfhwastoldev/telematic_cloud_messaging:${{ github.ref_name }}
+          tags: usdotfhwastolcandidate/telematic_cloud_messaging:${{ github.ref_name }}
+          build-args: |
+            BUILD_DATE=${{ steps.docker-image-metadata.outputs.build_date }}
```

Comment:
How are these build args used?

Reply:
as we are passing then during the docker build time process we can see the metadata details in the docker image layers once we build is finished so that the Docker hub image has metadata about when it was built, the commit it was built from.